packages/frontend/src/notebook/notebook_cell.tsx

```diff
@@ -134,11 +138,17 @@ export function NotebookCell(props: {
         },
     ];
 
+    const [closestEdge, setClosestEdge] = createSignal<ClosestEdge>(null);
+    const [dropTarget, setDropTarget] = createSignal(false);
+
     createEffect(() => {
         const cleanup = combine(
             draggable({
                 element: handleRef,
-                getInitialData: () => createCellDragData(props.cellId),
+                getInitialData: () => ({
+                    ...createCellDragData(props.cellId, props.index),
+                    index: props.index,
```

Comment:
If I'm not mistaken, this is redundant: your modified `createCellDragData` already adds the `index` field to the returned data. So I don't think you need to make any change here at all.